cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -202,12 +202,16 @@ func (s *CMServer) Run(_ []string) error {
 	}
 
 	if s.AllocateNodeCIDRs {
-		routes, ok := cloud.Routes()
-		if !ok {
-			glog.Fatal("Cloud provider must support routes if allocate-node-cidrs is set")
+		if cloud != nil {
```

Comment:
Should be in scope from the point of initialization through the rest of the if-else block.
My simple test here appears to work: http://play.golang.org/p/CwUM1SPgCK